pkg/controller/job_test.go

```diff
@@ -39,54 +41,93 @@ func TestKleaner_DeleteJob(t *testing.T) {
 			successful: time.Second,
 			failed:     time.Second,
 			ignoreCron: false,
 			expected:   false,
 		},
 		"jobs with active pods should not be deleted": {
-			jobSpec:    createJob(false, ts.Add(-time.Minute), 1, 0, 0), // job.Status.Active > 0
+			jobSpec:    createJob(false, ts.Add(-time.Minute), 1, 0, 0, []batchv1.JobCondition{}), // job.Status.Active > 0
 			successful: time.Second,
 			failed:     time.Second,
 			ignoreCron: false,
 			expected:   false,
 		},
 		"expired successful jobs should be deleted": {
-			jobSpec:    createJob(false, ts.Add(-time.Minute), 0, 1, 0),
+			jobSpec:    createJob(false, ts.Add(-time.Minute), 0, 1, 0, []batchv1.JobCondition{}),
 			successful: time.Second,
 			failed:     time.Second,
 			ignoreCron: false,
 			expected:   true,
 		},
 		"non-expired successful jobs should not be deleted": {
-			jobSpec:    createJob(false, ts.Add(-time.Minute), 0, 1, 0),
+			jobSpec:    createJob(false, ts.Add(-time.Minute), 0, 1, 0, []batchv1.JobCondition{}),
 			successful: time.Minute * 2,
 			failed:     time.Second,
 			ignoreCron: false,
 			expected:   false,
 		},
 		"expired failed jobs should be deleted": {
-			jobSpec:    createJob(false, ts.Add(-time.Minute), 0, 0, 1),
+			jobSpec:    createJob(false, ts.Add(-time.Minute), 0, 0, 1, []batchv1.JobCondition{}),
 			successful: time.Second,
 			failed:     time.Second,
 			ignoreCron: false,
 			expected:   true,
 		},
 		"non-expired failed jobs should not be deleted": {
-			jobSpec:    createJob(false, ts.Add(-time.Minute), 0, 0, 1),
+			jobSpec:    createJob(false, ts.Add(-time.Minute), 0, 0, 1, []batchv1.JobCondition{}),
 			successful: time.Second,
 			failed:     time.Minute * 2,
 			ignoreCron: false,
 			expected:   false,
 		},
+		"DeadlineExceeded jobs should be deleted": {
+			jobSpec: createJob(false, time.Time{}, 0, 0, 0, []batchv1.JobCondition{
+				batchv1.JobCondition{
+					Type:               batchv1.JobFailed,
+					Status:             corev1.ConditionTrue,
+					LastProbeTime:      metav1.NewTime(ts),
+					LastTransitionTime: metav1.NewTime(ts.Add(-time.Minute)),
+					Reason:             "DeadlineExceeded",
+					Message:            "Job was active longer than specified deadline",
+				},
+			}),
+			successful: time.Second,
+			failed:     time.Second,
+			ignoreCron: false,
+			expected:   true,
+		},
+		"successful but 'active' jobs should be deleted": {
+			jobSpec: createJob(false, ts.Add(-time.Minute), 1, 1, 0, []batchv1.JobCondition{
+				batchv1.JobCondition{
+					Type:               batchv1.JobComplete,
+					Status:             corev1.ConditionTrue,
+					LastProbeTime:      metav1.NewTime(ts),
+					LastTransitionTime: metav1.NewTime(ts.Add(-time.Minute)),
+					Reason:             "",
+					Message:            "",
+				},
+			}),
+			successful: time.Second,
+			failed:     time.Second,
+			ignoreCron: false,
+			expected:   true,
+		},
+		"failed but 'active' jobs should be deleted": {
```

Comment:
shouldn't here be jobConditions? I don't quite understand this case.

Reply:
I specifically didn't add it here, because if we see `"failed": 1` — it's enough to distinguish the job as failed (I haven't seen failed job with `"active": 1`, that's why)

But you're right, adding `jobConditions` is another test case I can add — will do!